packages/geolocator/example/lib/main.dart

```diff
@@ -0,0 +1,414 @@
+import 'dart:async';
+import 'dart:io' show Platform;
+
+import 'package:baseflow_plugin_template/baseflow_plugin_template.dart';
+import 'package:flutter/material.dart';
+import 'package:geolocator/geolocator.dart';
+
+/// Defines the main theme color.
+final MaterialColor themeMaterialColor =
+    BaseflowPluginExample.createMaterialColor(
+        const Color.fromRGBO(48, 49, 60, 1));
+
+void main() {
+  runApp(const GeolocatorWidget());
+}
+
+/// Example [Widget] showing the functionalities of the geolocator plugin
+class GeolocatorWidget extends StatefulWidget {
+  /// Creates a new GeolocatorWidget.
+  const GeolocatorWidget({Key? key}) : super(key: key);
+
+  /// Utility method to create a page with the Baseflow templating.
+  static ExamplePage createPage() {
+    return ExamplePage(
+        Icons.location_on, (context) => const GeolocatorWidget());
+  }
+
+  @override
+  _GeolocatorWidgetState createState() => _GeolocatorWidgetState();
+}
+
+class _GeolocatorWidgetState extends State<GeolocatorWidget> {
+  static const String _kLocationServicesDisabledMessage =
+      'Location services are disabled.';
+  static const String _kPermissionDeniedMessage = 'Permission denied.';
+  static const String _kPermissionDeniedForeverMessage =
+      'Permission denied forever.';
+  static const String _kPermissionGrantedMessage = 'Permission granted.';
+
+  final GeolocatorPlatform _geolocatorPlatform = GeolocatorPlatform.instance;
+  final List<_PositionItem> _positionItems = <_PositionItem>[];
+  StreamSubscription<Position>? _positionStreamSubscription;
+  StreamSubscription<ServiceStatus>? _serviceStatusStreamSubscription;
+
+  @override
+  void initState() {
+    super.initState();
+    _toggleServiceStatusStream();
+  }
+
+  PopupMenuButton _createActions() {
+    return PopupMenuButton(
+      elevation: 40,
+      onSelected: (value) async {
+        switch (value) {
+          case 1:
+            _getLocationAccuracy();
+            break;
+          case 2:
+            _requestTemporaryFullAccuracy();
+            break;
+          case 3:
+            _openAppSettings();
+            break;
+          case 4:
+            _openLocationSettings();
+            break;
+          case 5:
+            setState(_positionItems.clear);
+            break;
+          default:
+            break;
+        }
+      },
+      itemBuilder: (context) => [
+        if (Platform.isIOS)
+          const PopupMenuItem(
+            child: Text("Get Location Accuracy"),
+            value: 1,
+          ),
+        if (Platform.isIOS)
+          const PopupMenuItem(
+            child: Text("Request Temporary Full Accuracy"),
+            value: 2,
+          ),
+        const PopupMenuItem(
+          child: Text("Open App Settings"),
+          value: 3,
+        ),
+        if (Platform.isAndroid)
+          const PopupMenuItem(
+            child: Text("Open Location Settings"),
+            value: 4,
+          ),
```

Comment:
You can remove `if (Platform.isAndroid)`.

Reply:
Should I remove?  I prefer to leave the original example as is.

Reply:
My recommendation is to remove it.